Fixes 1042: Clarify release tagging

pljones opened the pull request:
**Short description of changes**
Clarify release tagging instructions.

**Context: Fixes an issue? Related issues**
Fixes: 1042

**Status of this Pull Request**
Another point raised whilst I was going through the 3.11.0 release (very minor).

**What is missing until this pull request can be merged?**
Review.

**Does this need translation?**
The doc has translation copies.

## Checklist
<!-- Please tick the check boxes when done by replacing the space by an x, e.g. [x]. -->
- [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
- [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. <!-- GitHub doesn't run these checks for new contributors automatically. -->
- [x] I'm sure that this Pull Request goes to the correct branch

pljones committed:
Fixes 1042: Clarify release tagging

softins commented on contribute/en/Release-Process.md:
```diff
@@ -284,10 +284,10 @@ Current state: <!-- Planning|Translations (beta)|Code freeze (rc)|Released -->
-- [ ] Tag this commit as `r3_y_z`
-  - [ ] Wait for the build to complete
+- [ ] [Tag the release version](https://jamulus.io/contribute/Release-Process#steps-for-a-specific-release):
+  - [ ] Update the version number in `Jamulus.pro` and add the release date to the Changelog header and commit.
```
I think updating `Jamulus.pro` and `Changelog` need to be done and committed before tagging, otherwise the release will not contain the proper version and change log?

pljones replied:
The details seem duplicated in the referenced section.

ann0see replied:
I'd prefer to have it here clearly as this avoids jumping. So roughly:

1. Change Jamulus.pro and changelog version number
2. Commit this and tag it as rx_y_z
3. Add a new entry to the changelog and a update Jamulus.pro

pljones replied:
You mean everywhere we refer to `steps-for-a-specific-release`, copy the whole section in and get rid of that section, or what?  If you don't do what that section says, you won't do it right.

ann0see replied:
We should have the lowest friction as possible. So links are one step more. If we do link it, then also the "Update version number ..." stuff is unnecessary and we could just say "follow the release instructions on ..."

pljones replied:
There are two flows:
- tagging and releasing _any_ tagged release, be it dev, beta, rc or final release
- the additional steps necessary _only_ when tagging a final release

What I was trying to do was make that clearer.

pljones replied:
Latest version: http://jamuluswebsite.drealm.info/contribute/Release-Process#steps-for-a-specific-release
This probably still needs further work, to be honest.  It's referenced as "Tag the release version" here but covers mostly translations:
> #### 1. Ensure .ts files are up to date (...)
> #### 2. Notify all the translators that translation is required (...)
> #### 3. Update the .ts files returned by translators (...)
> #### 4. When all translations have been submitted and merged. (...) **<-- [This](http://jamuluswebsite.drealm.info/contribute/Release-Process#4-when-all-translations-have-been-submitted-and-merged) is where tagging starts**
>#### If this is a proper release, move the latest tag (...) *(this intended to be a sub-step of 4, I think)*

pljones replied:
Maybe steps 1-3 should get split into "Translation Process" and step 4 and its sub-step should be "Release Tagging Process"?

pljones replied:
Also, the section on updating the ChangeLog would be better placed, in the overall Release Process page, between the section about translations and the section on performing the release -- it needs to happen _before_ the final release and it never gets mentioned in the tagging section(s) (just repeatedly from here, in the places it needs to be).

pljones replied:
I also notice this doc doesn't appear to reference the steps in Release Process about the translations.

ann0see reviewed (approved):
I think it's ok (I've now seen that it's a subpoint...). Ideally, we'd have a release script/automation for the tagging procedure.

pljones commented:
Locking Weblate, then merging outstandings before I do this.

ann0see commented:
Hmm. Yes. You're right. There was a webpage PR open on while I merged the other PR. 😐

pljones commented:
Weblate merged, updated weblate, rebased this, merged this, updated weblate again (probably didn't need the first one...), unlocked weblate.

Hopefully that'll keep weblate calm...

pljones commented:
I've raised #1076 for [my other comments](https://github.com/jamulussoftware/jamuluswebsite/pull/1073#discussion_r1894908304).
